### PR TITLE
chore: DSW-1753 -  Update page structure for NextJS

### PR DESCRIPTION
**Pull request by siggerzz**

### CODE REVIEW

File: nextjs-app/src/pages/components/button.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import NavigationLayout from "@/layout/navigation";
+import { PieButton } from "@justeattakeaway/pie-button/dist/react";
+import { PieDivider } from "@justeattakeaway/pie-divider/dist/react";
+import { IconSearch } from "@justeattakeaway/pie-icons-webc/dist/react/IconSearch"
```
Comment (jamieomaguire) [inline nextjs-app/src/pages/components/button.tsx:4]: `@justeattakeaway/pie-icons-webc/react/IconSearch` Should work here. We set up proper paths for the icons library

Comment (siggerzz) [reply nextjs-app/src/pages/components/button.tsx:4]: Not the case unfortunately. It looks like we don't actually export that folder in the icons package:

```
  "name": "@justeattakeaway/pie-icons-webc",
  "version": "0.17.2",
  "type": "module",
  "main": "dist/index.js",
  "module": "dist/index.js",
  "files": [
    "dist"
  ],
  ```

[the remark this answers is not in the document]
Comment (siggerzz) [reply nextjs-app/src/pages/components/button.tsx:4]: ditto

Comment (siggerzz) [reply nextjs-app/src/pages/components/button.tsx]: As discussed, we're currently seeing this issue for Next - https://github.com/vercel/next.js/issues/39375

File: nextjs-app/src/layout/navigation.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    return (
+      <>
+        <Head>
+            { isHomePage && <title>PIE Aperture</title>}
+          {!isHomePage && <title>PIE Aperture | NextJS | { title }</title>}
+        </Head>
+        <h1>NextJS - PIE { title } </h1>
+        { !isHomePage && <PieLink onClick={() => router.push('/')} tag="button">Back to home</PieLink>}
+        { !isHomePage && <PieDivider/>}
```
Comment (raoufswe) [inline nextjs-app/src/layout/navigation.tsx:25]:
```suggestion
        { !isHomePage && 
        <>
        <PieLink onClick={() => router.push('/')} tag="button">Back to home</PieLink>
        <PieDivider/>
        </>
        }
```

File: nextjs-app/src/layout/navigation.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+export default function NavigationLayout({ children, title }: NavigationLayoutProps) {
+
+    const router = useRouter();
+    const isHomePage = router.pathname === '/';
+
+    return (
+      <>
+        <Head>
+            { isHomePage && <title>PIE Aperture</title>}
+          {!isHomePage && <title>PIE Aperture | NextJS | { title }</title>}
```
Comment (raoufswe) [inline nextjs-app/src/layout/navigation.tsx:21]:
```suggestion
            { isHomePage ? <title>PIE Aperture</title> : <title>PIE Aperture | NextJS | { title }</title>}
```

File: nextjs-app/src/pages/components/button.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import NavigationLayout from "@/layout/navigation";
+import { PieButton } from "@justeattakeaway/pie-button/dist/react";
+import { PieDivider } from "@justeattakeaway/pie-divider/dist/react";
+import { IconSearch } from "@justeattakeaway/pie-icons-webc/dist/react/IconSearch";
+import { useState } from "react";
+
+export default function Button() {
+
```
Comment (raoufswe) [inline nextjs-app/src/pages/components/button.tsx:8]: nit: let's remive this line and add it after the useState call

File: nextjs-app/src/pages/components/icon-button.tsx [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import NavigationLayout from "@/layout/navigation";
+import { PieIconButton } from "@justeattakeaway/pie-icon-button/dist/react"
+import { IconClose } from "@justeattakeaway/pie-icons-webc/dist/react/IconClose";
+
+export default function IconButton() {
+    return (
+        <NavigationLayout title="Icon Button">
+        <PieIconButton>
```
Comment (raoufswe) [inline nextjs-app/src/pages/components/icon-button.tsx:8]:
```suggestion
        <PieIconButton onClick={() => console.log("clicked")}>
```

File: nextjs-app/src/pages/components/switch.tsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import NavigationLayout from "@/layout/navigation";
+import { PieSwitch } from "@justeattakeaway/pie-switch/dist/react";
+import { useState } from "react";
+
+export default function Switch() {
+
```
Comment (raoufswe) [inline nextjs-app/src/pages/components/switch.tsx:6]: nit: let's remove the line here

**Pull request merged**